### ISSUE TITLE
jupyter-client 7.0+ support

### ISSUE THREAD

**Issue by rpigott** [WolframResearch/WolframLanguageForJupyter#118]
Backwards incompatible changes in jupyter-client 7.0+ have broken the jupyter console with this kernel.

The following traceback is printed when attempting to execute any statement:

```
Unhandled exception in event loop:
  File "/usr/lib/python3.9/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/lib/python3.9/site-packages/prompt_toolkit/input/vt100.py", line 170, in callback_wrapper
    callback()
  File "/usr/lib/python3.9/site-packages/prompt_toolkit/application/application.py", line 690, in read_from_input
    self.key_processor.process_keys()
  File "/usr/lib/python3.9/site-packages/prompt_toolkit/key_binding/key_processor.py", line 271, in process_keys
    self._process_coroutine.send(key_press)
  File "/usr/lib/python3.9/site-packages/prompt_toolkit/key_binding/key_processor.py", line 186, in _process
    self._call_handler(matches[-1], key_sequence=buffer[:])
  File "/usr/lib/python3.9/site-packages/prompt_toolkit/key_binding/key_processor.py", line 321, in _call_handler
    handler.call(event)
  File "/usr/lib/python3.9/site-packages/prompt_toolkit/key_binding/key_bindings.py", line 124, in call
    result = self.handler(event)
  File "/usr/lib/python3.9/site-packages/jupyter_console/ptshell.py", line 475, in _
    self.handle_iopub()
  File "/usr/lib/python3.9/site-packages/jupyter_console/ptshell.py", line 854, in handle_iopub
    msg_type = sub_msg['header']['msg_type']
```

jupyter client has a migration guide published which may help: https://jupyter-client.readthedocs.io/en/latest/migration.html.

The changes don't look too substantial, but I'm not too familiar with the jupyter architecture and wasn't able to correct the issue.

**Comment by cc-wr** [comment]
This seems like a general issue with `jupyter-console` when it is used with `jupyter-client` 7.0+. I got the same issue when I tried the IPython kernel on the console with `jupyter-client` 7.0+.

There are some other GitHub issues for this at https://github.com/jupyter/jupyter_client/issues/690 and https://github.com/jupyter/jupyter_console/issues/251. Thankfully, it seems like the `jupyter-console` people have merged in a fix for this bug at https://github.com/jupyter/jupyter_console/pull/244. I tried the changes myself, and they seemed to work:

```
$ jupyter --version | grep jupyter_client
jupyter_client   : 7.1.0
$ jupyter console --kernel=wolframlanguage13
Jupyter console 6.4.0 # with the changes from #244 manually added in

Wolfram Language/Wolfram Engine Copyright 2019

Note: This Jupyter kernel was installed through the WolframScript install method. Accordingly, updates to a WolframLanguageForJupyter paclet will not affect this kernel.
In [1]: 2+2
Out[1]: 4

```

I think we just have to wait for a released version of `jupyter-console` with these changes. For now, a temporary solution would be to just downgrade the version of `jupyter-client` of course.
